consensus/scheme/rolldpos/fsm.go

```diff
@@ -539,18 +560,33 @@ func (m *cFSM) handleEndorseProposalEvt(evt fsm.Event) (fsm.State, error) {
 	}
 	endorsementSet.AddEndorsement(endorse)
 	validNum := endorsementSet.NumOfValidEndorsements(
+		expectedConsensusTopics,
+		m.ctx.epoch.delegates,
+	)
+	if validNum <= len(m.ctx.epoch.delegates)*2/3 {
+		return nil, nil, nil
+	}
+	return endorsementSet, nil, nil
+}
+
+func (m *cFSM) handleEndorseProposalEvt(evt fsm.Event) (fsm.State, error) {
+	endorsementSet, err, panicErr := m.processEndorseEvent(
+		evt,
+		eEndorseProposal,
 		map[endorsement.ConsensusVoteTopic]bool{
 			endorsement.PROPOSAL: true,
 			endorsement.COMMIT:   true, // commit endorse is counted as one proposal endorse
 		},
-		m.ctx.epoch.delegates,
 	)
-	if validNum <= len(m.ctx.epoch.delegates)*2/3 {
-		// Wait for more lock votes to come
-		return sAcceptProposalEndorse, nil
+	if panicErr != nil {
```

Comment:
sysErr?

Reply:
And should we define a ErrSys. processEndorseEvent only returns one error. If errors.Cause(err) == ErrSys -> blah blah